code/modules/projectiles/gun_helpers.dm

```diff
@@ -861,6 +861,34 @@ should be alright.
 		if((GUN_FIREMODE_AUTOMATIC in gun_firemode_list) && !(GUN_FIREMODE_AUTOBURST in gun_firemode_list))
 			add_firemode(GUN_FIREMODE_AUTOBURST, user)
 
+/obj/item/weapon/gun/proc/toggle_aim_mode(mob/living/carbon/human/user)
+	if(CHECK_BITFIELD(flags_gun_features, GUN_IS_AIMING))
+		gun_iff_signal = list()
+		DISABLE_BITFIELD(flags_gun_features, GUN_IS_AIMING)
+		user.remove_movespeed_modifier(MOVESPEED_ID_AIM_MODE_SLOWDOWN)
+		modify_fire_delay(-aim_fire_delay_buffer)
+		to_chat(user, "<span class='notice'>You seize aiming.</b></span>")
+		return
+	else if(!CHECK_BITFIELD(flags_item, WIELDED))
+		to_chat(user, "<span class='notice'>You need to wield your gun before aiming.</b></span>")
+		return
+	else if(!user.wear_id)
+		to_chat(user, "<span class='notice'>You don't have distinguished allies you want to avoid shooting.</b></span>")
+		return
+	else
+		to_chat(user, "<span class='notice'>You steady your breathing...</b></span>")
```

Comment:
```suggestion
	if(!CHECK_BITFIELD(flags_item, WIELDED))
		to_chat(user, "<span class='notice'>You need to wield your gun before aiming.</b></span>")
		return
	if(!user.wear_id)
		to_chat(user, "<span class='notice'>You don't have distinguished allies you want to avoid shooting.</b></span>")
		return
	to_chat(user, "<span class='notice'>You steady your breathing...</b></span>")
```